sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonParser.scala

```diff
@@ -34,6 +34,8 @@ import org.apache.spark.util.Utils
 
 private[sql] class SparkSQLJsonProcessingException(msg: String) extends RuntimeException(msg)
 
+private[sql] class NotAllowedNullException(msg: String) extends SparkSQLJsonProcessingException(msg)
```

Comment:
I added this one to differentiate this case from other `SparkSQLJsonProcessingException`[1] because when it fails to parse due to not-allowed null, `permissive` parse `mode` is not allowed (because `permissive` mode virtually means allowing `null`s on other fields).

[1]https://github.com/HyukjinKwon/spark/blob/cb5ece70bc840820697fa73943bc28bd63c96b41/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonParser.scala#L471-L474